bundle/artifacts/build.go

```diff
@@ -27,61 +29,146 @@ func (m *build) Name() string {
 func (m *build) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
 	var diags diag.Diagnostics
 
+	cacheDir, err := b.CacheDir(ctx)
```

Comment:
Maybe it's better to move it closer to where it's used later in the code?

Reply:
I thought about this, but then I need to cache the result to avoid calling the function multiple times. 

I don't think there we support the situation where CacheDir has failed but we want to continue with the build/deploy (let me know otherwise), so it's okay to do it preemptively.

Reply:
> I don't think there we support the situation where CacheDir has failed

Hm, but now we _do_ continue with the build if cache dir error (no return in if clause above), is it intentional then?

Reply:
You're right. The code is intentional, my comment is wrong. 

I do continue if cacheDir has failed as I want to wheel patching feature to be fail open.

I do it early so that I only call it at most once.

Reply:
We need the cache dir elsewhere too, so it makes sense to initialize it once and early, and then use it as a regular field on the bundle struct instead of a function call.